core/src/main/java/fr/sncf/osrd/envelope_sim/PhysicsRollingStock.java

```diff
@@ -34,17 +34,32 @@ public interface PhysicsRollingStock {
 
     /** Get the effort the train can apply at a given speed, in newtons */
     static double getMaxEffort(double speed, TractiveEffortPoint[] tractiveEffortCurve) {
-        double previousEffort = 0.0;
-        double previousSpeed = 0.0;
-        for (var dataPoint : tractiveEffortCurve) {
-            if (previousSpeed <= Math.abs(speed) && Math.abs(speed) < dataPoint.speed()) {
-                var coeff = (previousEffort - dataPoint.maxEffort()) / (previousSpeed - dataPoint.speed());
-                return previousEffort + coeff * (Math.abs(speed) - previousSpeed);
+        int index = 0;
+        int left = 0;
+        int right = tractiveEffortCurve.length - 1;
+        while (left <= right) {
+            int mid = (left + right) >>> 1;
```

Comment:
I assume this is to avoid overflows when computing the average? I didn't know this worked. I think a comment here would be helpful.